### PR TITLE
rsx: tiny zcull optimizations

### PR DESCRIPTION
**Pull request by scribam**
~~Should boost the performance on heavy zcull games.~~

### CODE REVIEW

**Comment by vietnam13231** [comment]
Infamous still needs to have "Disable ZCull Occlusion Queries" enabled for those to not overflow.

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 1]
```diff
@@ -2629,7 +2636,7 @@ namespace rsx
 		}
 
 		zcull_ctrl->set_enabled(this, zcull_rendering_enabled);
-		zcull_ctrl->set_active(this, zcull_rendering_enabled && testing_enabled && zcull_surface_active);
+		zcull_ctrl->set_active(this, zcull_surface_active);
```
Comment (kd-11) [inline rpcs3/Emu/RSX/RSXThread.cpp]: Make zcull_surface_active a local variable as it no longer represents a global status. It is possible to have a zcull surface active even with tests disabled.

**Comment by Xcedf** [comment]
None of my zcull needing games react on this PR

**Comment by MSuih** [comment]
https://streamable.com/sr66x

**Comment by MSuih** [comment]
Burnout: Paradise has too agressive culling, which results in models popping in or appearing gray. Hard to show on picture, so here's a video: https://streamable.com/sr66x

Master and PS3 do not have any noticeable pop-in.

**Comment by kd-11** [comment]
Thank you @MSuih, I have also confirmed your findings. That helped figure out why there were performance gains; it was because zcull testing was illegally disabled.
@scribam 
WRT the changes in check_zcull_status.
Picture the following:
1. The game changes the current depth buffer to some memory that is linked to the ZCULL unit. Zcull statistics are not enabled so no testing is expected. Your optimization aborts early and no ZCULL usage takes place.
2. The game enables Zcull testing. There is no framebuffer swapped however, so the set_active call now fails since zcull_surface_active was never updated and remains forever false.

**Comment by scribam** [comment]
Changes related to the Burnout regression reverted

**Pull request merged**